### PR TITLE
Add ANTLR4

### PR DESCRIPTION
**Pull request by dsogari**
ANTLR4 C++ runtime
https://github.com/DanMcLaughlin/antlr4/tree/master/runtime/Cpp

### CODE REVIEW

**Comment by Alexpux** [comment]
Is it not conflict with v3?

**Comment by dsogari** [comment]
I think not, because the library name for this one is ```libantlr4-runtime.dll``` while ANTLR3 is ```libantlr3c.dll```. BTW, I need to adjust the structure of the installation directories, to make the dll go to the bin folder. I'll update the patch.

**Comment by Alexpux** [comment]
What about headers and import libraries conflict?

**Comment by dsogari** [comment]
Yes, perhaps the include directory needs to be adjusted too. I'll see to it.

All the library files are named ```libantlr4-runtime.*```. Do you think maybe that could cause problems?

File: mingw-w64-antlr4-git/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}-git"
+pkgver=4.5.3.r4904.6f62821
+pkgrel=1
+pkgdesc="ANother Tool for Language Recognition v4 (mingw-w64)"
+arch=('any')
+url="http://www.antlr.org/"
+license=('BSD')
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc" "${MINGW_PACKAGE_PREFIX}-cmake")
+source=("${_realname}::git+https://github.com/DanMcLaughlin/antlr4.git"
+        "${_realname}-${pkgver}.patch")
```
Comment (mati865) [inline mingw-w64-antlr4-git/PKGBUILD]: The patch cannot use `${_realname}-${pkgver}` format because it will break with update.

**Pull request merged**